activesupport/lib/active_support/core_ext/string/inflections.rb

```diff
@@ -52,9 +56,49 @@ def pluralize(count = nil, locale = :en)
   #   'word'.singularize             # => "word"
   #   'the blue mailmen'.singularize # => "the blue mailman"
   #   'CamelOctopi'.singularize      # => "CamelOctopus"
+  #   'apples'.singularize(1)        # => "apple"
+  #   'apples'.singularize(2)        # => "apples"
   #   'leyes'.singularize(:es)       # => "ley"
-  def singularize(locale = :en)
-    ActiveSupport::Inflector.singularize(self, locale)
+  #   'leyes'.singularize(2, :es)    # => "leyes"
+  def singularize(count = 1, locale = :en)
+    count, locale = 1, count if count.is_a?(Symbol)
+    if count != 1
+      dup
+    else
+      ActiveSupport::Inflector.singularize(self, locale)
+    end
+  end
+
+  # Returns either the plural or the singular form of the word in the string.
+  #
+  # If the optional parameter +count+ is specified,
+  # the singular form will be returned if <tt>count == 1</tt>.
+  # For any other value of +count+ the plural will be returned.
+  #
+  # If the optional parameter +locale+ is specified,
+  # the word will be singularized as a word of that language.
+  # By default, this parameter is set to <tt>:en</tt>.
+  # You must define your own inflection rules for languages other than English.
+  #
+  #   'post'.with_count(1)             # => "post"
+  #   'posts'.with_count(1)            # => "post"
+  #   'post'.with_count(2)             # => "posts"
+  #   'posts'.with_count(2)            # => "posts"
+  #   'octopi'.with_count(2)           # => "octopus"
+  #   'sheep'.with_count(2)            # => "sheep"
+  #   'word'.with_count(2)             # => "word"
+  #   'the blue mailmen'.with_count(2) # => "the blue mailman"
```

Comment:
This is a bit confusing. How does pluralizing "The blue mailmen" end up as "The blue mailman"? So… two blue *mailmen(s)* is a blue mailman? What am I missing?